sync/sync.go

```diff
@@ -97,10 +107,114 @@ func (s *Synchronizer) WithListener(listener EventListener) *Synchronizer {
 
 // Run starts the Synchronizer, returns an error if the loop is already running
 func (s *Synchronizer) Run(ctx context.Context) error {
-	s.syncBlocks(ctx)
+	s.syncBlocks2(ctx)
 	return nil
 }
 
+func (s *Synchronizer) syncBlocks2(syncCtx context.Context) {
+	streamCtx, streamCancel := context.WithCancel(syncCtx)
+	i := s.nextHeight()
+	for ; ; i++ {
+		select {
+		case <-streamCtx.Done():
+			streamCancel()
+			return
+		default:
```

Comment:
If there is a failure in one of the statements below: we should immediately stop syncing (it's up to you how to resolve it `continue` or something else)

Reply:
[fixed](https://github.com/NethermindEth/juno/pull/1934/commits/f1846fce756bde5aa6e91f3062e38c9095865465)